core/banking/recurring_transfers.go

```diff
@@ -49,6 +49,27 @@ func (e *Engine) recurringTransfer(
 		return fmt.Errorf("could not transfer funds: %w", err)
 	}
 
+	if transfer.DispatchStrategy != nil {
+		hasAsset := len(transfer.DispatchStrategy.AssetForMetric) > 0
+		isMarketProposer := transfer.DispatchStrategy.Metric == vegapb.DispatchMetric_DISPATCH_METRIC_MARKET_VALUE
+
+		if !hasAsset && !isMarketProposer {
```

Comment:
this seems unnecessary because we're already checking this in checkTransfer

Reply:
fair enough.